packages/licensing/src/class-manager.php

```diff
@@ -0,0 +1,195 @@
+<?php
+/**
+ * A Terms of Service class for Jetpack.
+ *
+ * @package automattic/jetpack-licensing
+ */
+
+namespace Automattic\Jetpack\Licensing;
+
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+use Jetpack_IXR_ClientMulticall;
+use Jetpack_Options;
+use WP_Error;
+
+/**
+ * Class Manager.
+ *
+ * @since ??
+ *
+ * Helper class that is responsible for validating licenses to the current site.
+ */
+class Manager {
+	/**
+	 * Name of the WordPress option that holds all known Jetpack licenses.
+	 *
+	 * @const string
+	 */
+	const LICENSES_OPTION_NAME = 'jetpack_licenses';
+
+	/**
+	 * Holds the singleton instance of this class.
+	 *
+	 * @var self
+	 */
+	protected static $instance = false;
+
+	/**
+	 * Singleton.
+	 *
+	 * @static
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Initialize.
+	 *
+	 * @return void
+	 */
+	public function initialize() {
+		add_action( 'update_option_' . self::LICENSES_OPTION_NAME, array( $this, 'validate_stored_licenses' ) );
+		add_action( 'jetpack_authorize_ending_authorized', array( $this, 'validate_stored_licenses_on_connection' ) );
+	}
+
+	/**
+	 * Get Jetpack connection manager instance.
+	 *
+	 * @return Connection_Manager
+	 */
+	protected function connection() {
+		return new Connection_Manager();
+	}
+
+	/**
+	 * Make an authenticated WP.com API multicall request instance.
+	 *
+	 * @param array $client_args IXR client arguments.
+	 * @return Jetpack_IXR_ClientMulticall
+	 */
+	protected function request( $client_args ) {
+		return new Jetpack_IXR_ClientMulticall( $client_args );
```

Comment:
Of note, we now have `XMLRPC_Async_Call` that may simplify things a bit:

https://github.com/Automattic/jetpack/blob/80ceab350a36089f37134404aad47509f85df462/packages/connection/src/class-xmlrpc-async-call.php#L24

Reply:
Async doesn't work for us in this case as we need the responses in order to surface relevant errors to the user, if any.